lms/djangoapps/lti_provider/signature_validator.py

```diff
@@ -0,0 +1,244 @@
+"""
+Subclass of oauthlib's RequestValidator that checks an OAuth signature.
+"""
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from oauthlib.oauth1 import SignatureOnlyEndpoint
+from oauthlib.oauth1 import RequestValidator
+
+from lti_provider.models import LtiConsumer
+
+
+class SignatureValidator(RequestValidator):
+    """
+    Helper class that verifies the OAuth signature on a request.
+
+    The pattern required by the oauthlib library mandates that subclasses of
+    RequestValidator contain instance methods that can be called back into in
+    order to fetch the consumer secret or to check that fields conform to
+    application-specific requirements.
+    """
+
+    def __init__(self):
+        super(SignatureValidator, self).__init__()
+        self.endpoint = SignatureOnlyEndpoint(self)
+
+    # The OAuth signature uses the endpoint URL as part of the request to be
+    # hashed. By default, the oauthlib library rejects any URLs that do not
+    # use HTTPS. We turn this behavior off in order to allow edX to run without
+    # SSL in development mode. When the platform is deployed and running with
+    # SSL enabled, the URL passed to the signature verifier must start with
+    # 'https', otherwise the message signature would not match the one generated
+    # on the platform.
+    enforce_ssl = False
+
+    def check_client_key(self, key):
+        """
+        Verify that the key supplied by the LTI consumer is valid for an LTI
+        launch. This method is only concerned with the structure of the key;
+        whether the key is associated with a known LTI consumer is checked in
+        validate_client_key. This method signature is required by the oauthlib
+        library.
+
+        :return: True if the client key is valid, or False if it is not.
+        """
+        return key is not None and 0 < len(key) <= 32
```

Comment:
This is perfectly good as is, but just because you mentioned that Python's not your main language, I just wanted to make sure you knew that `None` and empty strings/collections all evaluate to false values, so you could also have done:

```python
return key and len(key) <= 32
```